### PR TITLE
Update AWS SDK to v2

### LINKED ISSUE #1054: Migrate to aws-sdk-go v2
**How to categorize this issue?**

/area quality productivity
/kind enhancement
/platform aws

**What would you like to be added**:
Migrate usage of v1 aws-go-sdk to v2.

**Why is this needed**:
[aws-sdk-go v1](https://github.com/aws/aws-sdk-go) is in maintenance mode. EOF will be reached on 31.7.2025. In the context of https://github.com/gardener/gardener/issues/9586 it would be great if we can directly implement the new functionalities with the [aws-sdk-go v2](https://github.com/aws/aws-sdk-go-v2) which provides a nicer API for working with JWTs https://github.com/aws/aws-sdk-go-v2/blob/f1d71c59a1499981873f70db8c92d07242779670/credentials/stscreds/web_identity_provider.go#L70

### PR DESCRIPTION
**Pull request by AndreasBurger**
**How to categorize this PR?**

/area quality
/kind task
/platform aws

**What this PR does / why we need it**:
This PR removes all usage of the old v1 SDK and replaces it with v2

**Which issue(s) this PR fixes**:
Fixes #1054

**Special notes for your reviewer**:
No sense in reviewing before it is undrafted. This is heavily WIP :)

**Release note**:

```noteworthy developer
Updated AWS SDK from v1 to v2
```

### CODE REVIEW

**Comment by AndreasBurger** [comment]
Current todos:
- restore methods using pagination
- determine proper replacement for some error handling during tests
- test

**Comment by AndreasBurger** [comment]
Undrafting this as everything checks out ok. There are probably some inconsistencies in usage of the new SDK types (e.g. using "enum" values directly vs using them stringified).

File: pkg/controller/infrastructure/configvalidator.go [hunk 1]
```diff
@@ -91,7 +92,7 @@ func (c *configValidator) validateVPC(ctx context.Context, awsClient awsclient.I
 	allErrs := field.ErrorList{}
 
 	// Verify that the VPC exists and the enableDnsSupport and enableDnsHostnames VPC attributes are both true
-	for _, attribute := range []string{"enableDnsSupport", "enableDnsHostnames"} {
+	for _, attribute := range []string{string(ec2types.VpcAttributeNameEnableDnsSupport), string(ec2types.VpcAttributeNameEnableDnsHostnames)} {
 		value, err := awsClient.GetVPCAttribute(ctx, vpcID, attribute)
```
[the remark this answers is not in the document]
Comment (AndreasBurger) [reply pkg/controller/infrastructure/configvalidator.go:96]: Not sure I got that, golang kept complaining at every change I tried. I've adjusted this place and the function called (seemed to be used only here) to just use the aliased type directly.

File: pkg/aws/matchers/matchers.go [hunk 2]
```diff
@@ -96,10 +96,10 @@ func genericConsistOfSemanticallyEqual(expected interface{}) types.GomegaMatcher
 
 	for i := 0; i < value.Len(); i++ {
 		expectedElement := value.Index(i)
-		if expectedElement.IsNil() {
-			expectedElements = append(expectedElements, BeNil())
-		} else {
+		if value.Kind() != reflect.Pointer {
```
[the remark this answers is not in the document]
Comment (AndreasBurger) [reply pkg/aws/matchers/matchers.go]: Iirc, we did an intermediate change way back when (which did not fix the issue with the matcher here, a premature ptr-deref, I think) and then I _think_ I realised we just need to switch the order of comparisons. So this is probably a leftover 🤔.
Either way, I've changed it to only switch the order of comparisons and the tests ran successfully locally.

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**